src/client/QXmppAccountMigrationManager.h

```diff
@@ -57,13 +57,13 @@ class QXMPP_EXPORT QXmppExportData
         registerExtensionInternal(std::type_index(typeid(T)), parseAny, serializeAny, tagName, xmlns);
     }
 
+    const std::unordered_map<std::type_index, std::any> &extensions() const;
+    void setExtension(std::any value);
+
 private:
     friend class QXmppAccountMigrationManager;
     friend class tst_QXmppAccountMigrationManager;
 
-    const std::unordered_map<std::type_index, std::any> &extensions() const;
-    void setExtension(std::any value);
-
```

Comment:
Can you keep those functions private and add the following (public):

```
template<typename T>
std::optional<T> extensionData() const;

template<typename T>
void setExtensionData(T &&);
```

(those functions maybe need private helpers, not sure)